core/trino-main/src/main/java/io/trino/execution/SqlTaskExecution.java

```diff
@@ -569,9 +602,16 @@ public DriverSplitRunner createDriverRunner(@Nullable ScheduledSplit partitioned
             return new DriverSplitRunner(this, driverContext, partitionedSplit);
         }
 
+        @Nullable
         public Driver createDriver(DriverContext driverContext, @Nullable ScheduledSplit partitionedSplit)
         {
+            if (taskStateMachine.getState().isTerminatingOrDone()) {
+                return null;
+            }
             Driver driver = driverFactory.createDriver(driverContext);
+            // Register driver destroyed listener to detect when termination completes
+            liveCreatedDrivers.incrementAndGet();
+            driver.getDestroyedFuture().addListener(driverAndTaskTerminationTracker.getDriverDestroyedListener(), directExecutor());
```

Comment:
nit: Remove `getDriverDestroyedListener` and pass the `driverAndTaskTerminationTracker` directly

Reply:
Done